packages/calling-stateful-client/src/CallContext.ts

```diff
@@ -415,7 +417,7 @@ export class CallContext {
     this.modifyState((draft: CallClientState) => {
       const call = draft.calls[this._callIdHistory.latestCallId(callId)];
 
-      if (!call) {
+      if (!call || !isCapableToReact(call.capabilitiesFeature)) {
```

Comment:
If I'm not capable to react, I think we should still be able to see others' reactions?

Reply:
(e.g. if I can't unmute or share video, I can still hear/see others)

Reply:
When Capability is turned off in a meeting, None can react. Hence local participant will not receive any reactions as well. 
Since, Web SDK establishes a socket connection through the live state service it can still propagate reaction messages since the Live State Service is unaware of the capability changes or updates. 

So in the teams interop scenario, users joining with UI library or their own acs app which uses SDK APIs, with wrong implementation can pass reaction message. So clients using teams app wont see the reactions but CTE's using UI lib will be able to see the reaction which might become a privacy concern. To have parity, implying this check. 

Web SDK does handle capability changes. But this can happen in Teams Group Calls since Web SDK's teams group call scenarios & 1 to 1 call handling with capability is not stable yet.

Reply:
> with wrong implementation can pass reaction message .... CTE's using UI lib will be able to see the reaction which might become a privacy concern.

Developers won't have this level of knowledge when building their own apps direct from the SDK to know they should be ignoring inbound reactions if the capability is set. If this is in any way a privacy issue we must handle this at the SDK level. What's the limitation of the SDK performing the same check here on the capability and not sending the reaction?

Reply:
Yes I agree. The feature in SDK checks for Capability while send or receiving reaction. 
The capability API does not handle cases for 1 to 1 & 1 to N teams group calls. We have handled these cases without capability API and made it stable. 
With more changing of Capability feature and API and removing direct tscall event listening from the feature, will make it problematic.